django_nyt/conf.py

```diff
@@ -0,0 +1,149 @@
+    Keys can have a glob pattern, like ``USER_*`` or ``user/*``.
+
+    When notification emails are generated,
+    they are grouped by their templates such that notifications sharing the same template can be sent in a combined email.
+
+    Example:
+
+    .. code-block:: python
+
+        NYT_EMAIL_TEMPLATE_NAMES = OrderedDict(
+            [
+                ("admin/product/created", "myapp/notifications/email/admin_product_added.txt"),
+                ("admin/*", "myapp/notifications/email/admin_default.txt"),
+            ]
+        )
+    """
+
+    NYT_EMAIL_SUBJECT_TEMPLATE_NAMES: dict = field(default_factory=OrderedDict)
+    """Default dictionary, mapping notification keys to template names. The templates are used to generate a single-line email subject.
+    Can be overwritten by database values.
+    Keys can have a glob pattern, like ``USER_*`` or ``user/*``.
+
+    When notification emails are generated,
+    they are grouped by their templates such that notifications sharing the same template can be sent in a combined email.
+
+    Example:
+
+    .. code-block:: python
+
+        NYT_EMAIL_SUBJECT_TEMPLATE_NAMES = OrderedDict(
+            [
+                ("admin/product/created", "myapp/notifications/email/admin_product_added.txt"),
+                ("admin/*", "myapp/notifications/email/admin_default.txt"),
+            ]
+        )
+    """
+
+    NYT_INTERVALS: list[tuple[int, Any]] | tuple[tuple[int, Any]] = (
+        (INSTANTLY, _("instantly")),
+        (DAILY, _("daily")),
+        (WEEKLY, _("weekly")),
+    )
+    """List of intervals available for user selections. In minutes"""
+
+    NYT_INTERVALS_DEFAULT: int = INSTANTLY
+    """Default selection for new subscriptions"""
+
+    NYT_USER_MODEL: str = getattr(django_settings, "AUTH_USER_MODEL", "auth.User")
+    """The swappable user model of Django Nyt. The default is to use the contents of ``AUTH_USER_MODEL``."""
+
+    ############
+    # CHANNELS #
+    ############
+
+    NYT_ENABLE_CHANNELS: str = (
+        "channels" in django_settings.INSTALLED_APPS
+        and not getattr(django_settings, "NYT_CHANNELS_DISABLE", False)
+    )
+    """Channels are enabled automatically when 'channels' application is installed,
+    however you can explicitly disable it with NYT_CHANNELS_DISABLE."""
+
+    # Name of the global channel (preliminary stuff) that alerts everyone that there
+    # is a new notification
+    NYT_NOTIFICATION_CHANNEL: str = "nyt_all-{notification_key:s}"
+
+    def __getattribute__(self, __name: str) -> Any:
+        """
+        Check if a Django project settings should override the app default.
+
+        In order to avoid returning any random properties of the django settings, we inspect the prefix firstly.
+        """
+
+        if __name.startswith(settings_prefix) and hasattr(django_settings, __name):
+            return getattr(django_settings, __name)
+
+        return super().__getattribute__(__name)
+
+
+app_settings = AppSettings()
```

Comment:
You could still define the `__getattr__` function here as in the first commit like so:

```python
def __getattr__(name: str) -> Any:
    # See https://peps.python.org/pep-0562/
    return getattr(app_settings, name)
```

This way you can do `from .conf import NOTIFICATION_CHANNEL`

Maybe a small thing which is not worth it, but it makes the module more opaque and doesn't tie the implementation to having a `app_settings` object to fetch the configuration from.

Reply:
Ooooh interesting! But I think it's too creative for IDE auto-suggestions?

Reply:
Hmm yeah you are probably right - just tested in PyCharm and it does not suggest the configuration options. Although I would think it should be able to: https://youtrack.jetbrains.com/issue/PY-28017/Support-module-getattr-and-dir-from-PEP-562

Reply:
Not a bad challenge to eventually solve... It's a similar issue if we move settings into AppConfig because it works through `apps.get_app_config("appname")`... maybe some helper functions with typehints to patch things up... but feels hacky.